weave/cookbooks/source/Intro_to_Weave_Hello_Eval.ipynb

```diff
@@ -56,124 +53,187 @@
    },
    "outputs": [],
    "source": [
-    "# Install dependancies and imports\n",
-    "!pip install wandb weave openai -q\n",
+    "# Find your wandb API key at: https://wandb.ai/authorize\n",
+    "# Ensure that your wandb API key is available at:\n",
+    "# os.environ['WANDB_API_KEY'] = \"<your_wandb_api_key>\"\n",
     "\n",
     "import os\n",
-    "from getpass import getpass\n",
+    "import re\n",
+    "from textwrap import dedent\n",
     "\n",
     "from openai import OpenAI\n",
-    "from pydantic import BaseModel\n",
     "\n",
     "import weave\n",
     "\n",
-    "# 🔑 Setup your API keys\n",
-    "# Running this cell will prompt you for your API key with `getpass` and will not echo to the terminal.\n",
-    "#####\n",
-    "print(\"---\")\n",
-    "print(\n",
-    "    \"You can find your Weights and Biases API key here: https://wandb.ai/settings#api\"\n",
-    ")\n",
-    "os.environ[\"WANDB_API_KEY\"] = getpass(\"Enter your Weights and Biases API key: \")\n",
-    "print(\"---\")\n",
-    "print(\"You can generate your OpenAI API key here: https://platform.openai.com/api-keys\")\n",
-    "os.environ[\"OPENAI_API_KEY\"] = getpass(\"Enter your OpenAI API key: \")\n",
-    "print(\"---\")\n",
-    "#####\n",
-    "\n",
-    "# 🏠 Enter your W&B project name\n",
-    "weave_client = weave.init(\"MY_PROJECT_NAME\")  # 🐝 Your W&B project name"
+    "# Setup your wandb project (team name / project name) 🐝\n",
+    "WANDB_ENTITY_PROJECT = \"my-team-name/my-project-name\"\n",
+    "\n",
+    "\n",
+    "class JsonModel(weave.Model):\n",
+    "    prompt: weave.Prompt = weave.StringPrompt(\n",
+    "        dedent(\"\"\"\n",
+    "You are an assistant that answers questions about JSON data provided by the user. The JSON data represents structured information of various kinds, and may be deeply nested. In the first user message, you will receive the JSON data under a label called 'context', and a question under a label called 'question'. Your job is to answer the question with as much accuracy and brevity as possible. Give only the answer with no preamble. You must output the answer in XML format, between <answer> and </answer> tags.\n",
+    "\"\"\")\n",
+    "    )\n",
+    "    model: str = \"OpenPipe/Qwen3-14B-Instruct\"\n",
+    "    _client: OpenAI\n",
+    "\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self._client = OpenAI(\n",
+    "            base_url=\"https://api.inference.wandb.ai/v1\",\n",
```

Comment:
Or is it the Inference example after all? (Maybe this is me not understanding the example, apologies if so)